### ISSUE TITLE
Triton Batch size feature requests

### ISSUE THREAD

**Issue by klappec-bsci** [triton-inference-server/server#4363]
**Is your feature request related to a problem? Please describe.**

I want to send batches of say, 500 images in a single HTTP request, even when the model only supports a batch size of say, 16. Why can't Triton simply break these up for me? Instead I have to create a new model, preprocess.py, that handles this work.

I'd also like the ability to send batches of differing sizes. For my usecase, you have to send the model checksum as part of the request, to make *sure* you're requesting the correct model. but if you have 500 images, you have to send 500 checksums! 
You should really only have to send in (1) checksum Tensor.

**Describe the solution you'd like**

The ability to send nonequal batches and batches larger than the model max batch size.

**Describe alternatives you've considered**
Current solution is to send in 500 images, 500 identical checksums, and to have a preprocess model.py break them up. OR to force the client software to break them up itself, and to collate the results.

**Comment by nv-kmcgill53** [comment]
Hi @klappec-bsci, thank you for your request. We will file a ticket for it.

**Comment by klappec-bsci** [comment]
**Thank you so much. Are there any current solutions to this I missed?**  If someone wants to send 500 images to a model that accepts batch size 8, they really do have to break their requests up?

and they really do have to send in 500 verification checksums, in this use case?

**Comment by nv-kmcgill53** [comment]
> If someone wants to send 500 images to a model that accepts batch size 8, they really do have to break their requests up?

Yes, Triton has features such as [BLS](https://github.com/triton-inference-server/python_backend#business-logic-scripting) and [Decoupled Mode](https://github.com/triton-inference-server/python_backend#decoupled-mode-beta) to help the user with this limitation.
